### ISSUE TITLE
Pip install issue, module not found

### ISSUE THREAD

**Issue by Andde** [BjoernLudwig/pyxml2pdf#8]
Hello,

Crashed a wall using python 3.9.0.
pip install seems to be ok
After that, running:
python -m pyxml2pdf.main input/template.xml
C:\Users\andde\AppData\Local\Programs\Python\Python39\python.exe: Error while finding module specification for 'pyxml2pdf.main' (ModuleNotFoundError: No module named 'pyxml2pdf')

**Comment by BjoernLudwigPTB** [comment]
I cannot recreate your issue with the provided input. For me the following test was successful:

```shell
$ python3.9 -m venv pyxml2pdf-py39                                   
$ source pyxml2pdf-py39/bin/activate             
$ pip install pyxml2pdf           
Collecting pyxml2pdf
  Downloading pyxml2pdf-0.3.2-py3-none-any.whl (394 kB)
     |████████████████████████████████| 394 kB 703 kB/s 
Collecting defusedxml
  Using cached defusedxml-0.7.1-py2.py3-none-any.whl (25 kB)
Collecting reportlab
  Downloading reportlab-3.6.6-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (2.8 MB)
     |████████████████████████████████| 2.8 MB 3.3 MB/s 
Collecting pypdf2
  Downloading PyPDF2-1.26.0.tar.gz (77 kB)
     |████████████████████████████████| 77 kB 699 kB/s 
Collecting download
  Downloading download-0.3.5-py3-none-any.whl (8.8 kB)
Collecting tqdm
  Using cached tqdm-4.62.3-py2.py3-none-any.whl (76 kB)
Collecting six
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting requests
  Using cached requests-2.27.1-py2.py3-none-any.whl (63 kB)
Collecting pillow>=4.0.0
  Downloading Pillow-9.0.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.3 MB)
     |████████████████████████████████| 4.3 MB 3.5 MB/s 
Collecting urllib3<1.27,>=1.21.1
  Using cached urllib3-1.26.8-py2.py3-none-any.whl (138 kB)
Collecting idna<4,>=2.5
  Using cached idna-3.3-py3-none-any.whl (61 kB)
Collecting charset-normalizer~=2.0.0
  Using cached charset_normalizer-2.0.11-py3-none-any.whl (39 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2021.10.8-py2.py3-none-any.whl (149 kB)
Using legacy 'setup.py install' for pypdf2, since package 'wheel' is not installed.
Installing collected packages: urllib3, idna, charset-normalizer, certifi, tqdm, six, requests, pillow, reportlab, pypdf2, download, defusedxml, pyxml2pdf
    Running setup.py install for pypdf2 ... done
Successfully installed certifi-2021.10.8 charset-normalizer-2.0.11 defusedxml-0.7.1 download-0.3.5 idna-3.3 pillow-9.0.1 pypdf2-1.26.0 pyxml2pdf-0.3.2 reportlab-3.6.6 requests-2.27.1 six-1.16.0 tqdm-4.62.3 urllib3-1.26.8
WARNING: You are using pip version 21.2.4; however, version 22.0.3 is available.
You should consider upgrading via the 'python3.9 -m pip install --upgrade pip' command.
$ python -m pyxml2pdf.main input/template.xml
No custom setting found for 'pagesize', falling back to (90.1, 84.3).
No custom setting found for 'rows_xmltag', falling back to row_tag.
No custom setting found for 'identifier_xmltag', falling back to ['name_tag', 'info_tag', 'filter_tag'].
No custom setting found for 'sort_xmltag', falling back to name_tag.
No custom setting found for 'columns', falling back to [Column(label='name', tag=['name_tag'], width=30), Column(label='info', tag=['info_tag'], width=30), Column(label='filter', tag=['filter_tag'], width=30)].
No custom setting found for 'filter_xmltag', falling back to filter_tag.
No custom setting found for 'subtable_settings', falling back to (SubtableSetting(label='filter 1', include=[['filter_1']]), SubtableSetting(label='filter 1 and filter 2', include=[['filter_1'], ['filter_2']]), SubtableSetting(label='filter 1 or filter 2', include=[['filter_1', 'filter_2']]), SubtableSetting(label='filter 2', include=[['filter_2']]), SubtableSetting(label='filter 2 and filter 3', include=[['filter_2'], ['filter_3']]), SubtableSetting(label='filter 2 or filter 3', include=[['filter_2', 'filter_3']]), SubtableSetting(label='filter 3', include=[['filter_3']]), SubtableSetting(label='filter 1 and filter 3', include=[['filter_1'], ['filter_3']]), SubtableSetting(label='filter 1 or filter 3', include=[['filter_1', 'filter_3']])).
No custom setting found for 'font', falling back to Font(normal='LiberationSans-Regular.ttf', italic='LiberationSans-Italic.ttf', bold='LiberationSans-Bold.ttf', bolditalic='LiberationSans-BoldItalic.ttf').
No custom setting found for 'fontsize', falling back to FontSize(normal=6.5, table_heading=12, column_heading=6.5).
file_sizes: 100%|███████████████████████████████| 767/767 [00:00<00:00, 882kB/s]
Create 3 single paged PDFs.

You can find them concatenated at file:///pyxml2pdf/pyxml2pdf_output.pdf

-------------------------------DONE-------------------------------
```

**Comment by BjoernLudwigPTB** [comment]
I would assume, that your `pip install` installed the package into a different interpreter, than the one you are using when trying to run `python -m pyxml2pdf.main input/template.xml`. The package on PyPI to me seems to be running normally as expected.

**Comment by Andde** [comment]
C:\report_generator>pip install --upgrade --force-reinstall pyxml2pdf --ignore-installed
Collecting pyxml2pdf
  Using cached pyxml2pdf-0.3.2-py3-none-any.whl (394 kB)
Collecting reportlab
  Using cached reportlab-3.6.6-cp39-cp39-win_amd64.whl (2.3 MB)
Collecting download
  Using cached download-0.3.5-py3-none-any.whl (8.8 kB)
Collecting defusedxml
  Using cached defusedxml-0.7.1-py2.py3-none-any.whl (25 kB)
Collecting pypdf2
  Using cached PyPDF2-1.26.0.tar.gz (77 kB)
  Preparing metadata (setup.py) ... done
Collecting tqdm
  Using cached tqdm-4.62.3-py2.py3-none-any.whl (76 kB)
Collecting six
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting requests
  Using cached requests-2.27.1-py2.py3-none-any.whl (63 kB)
Collecting pillow>=4.0.0
  Using cached Pillow-9.0.1-cp39-cp39-win_amd64.whl (3.2 MB)
Collecting certifi>=2017.4.17
  Using cached certifi-2021.10.8-py2.py3-none-any.whl (149 kB)
Collecting idna<4,>=2.5
  Using cached idna-3.3-py3-none-any.whl (61 kB)
Collecting urllib3<1.27,>=1.21.1
  Using cached urllib3-1.26.8-py2.py3-none-any.whl (138 kB)
Collecting charset-normalizer~=2.0.0
  Using cached charset_normalizer-2.0.11-py3-none-any.whl (39 kB)
Collecting colorama
  Using cached colorama-0.4.4-py2.py3-none-any.whl (16 kB)
Using legacy 'setup.py install' for pypdf2, since package 'wheel' is not installed.
Installing collected packages: pypdf2, certifi, urllib3, six, pillow, idna, defusedxml, colorama, charset-normalizer, tqdm, requests, reportlab, download, pyxml2pdf
  Running setup.py install for pypdf2 ... done
Successfully installed certifi-2021.10.8 charset-normalizer-2.0.11 colorama-0.4.4 defusedxml-0.7.1 download-0.3.5 idna-3.3 pillow-9.0.1 pypdf2-1.26.0 pyxml2pdf-1.0 reportlab-3.6.6 requests-2.27.1 six-1.16.0 tqdm-4.62.3 urllib3-1.26.8

C:\report_generator>python
Python 3.9.0 (tags/v3.9.0:9cf6752, Oct  5 2020, 15:34:40) [MSC v.1927 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> quit()

C:\report_generator>python -m pyxml2pdf.main input/template.xml
C:\Users\anders.sutinen\AppData\Local\Programs\Python\Python39\python.exe: Error while finding module specification for 'pyxml2pdf.main' (ModuleNotFoundError: No module named 'pyxml2pdf')

**Comment by Andde** [comment]
Hmm, all other packages which i've installed via pip have been working as expected. virtual env would complicate a little bit now, env otherwise win10. 
Tried to confirm the pip+python with: 
python -m pip install --upgrade --force-reinstall pyxml2pdf --ignore-installed 

pip uninstall pyxml2pdf is also complaining: 
Found existing installation: PyXML2PDF 1.0
ERROR: Cannot uninstall 'PyXML2PDF'. It is a distutils installed project and thus we cannot accurately determine which files belong to it which would lead to only a partial uninstall.

**Comment by BjoernLudwigPTB** [comment]
> Found existing installation: PyXML2PDF 1.0

That's a surprising message, since there is no version 1.0 yet. With these installation and uninstallation routines you are fiddling with the system interpreter. I would assume for now, that `python -m pip uninstall [...]` would result in another outcome (actually work). 

The same holds for `python -m pip install`. Can you try that as a next step:

```shell
python -m pip install --upgrade pyxml2pdf
```

You could as well compare the output of `pip list` to `python -m pip list` to get an idea, if differet interpreters are involved.

**Comment by BjoernLudwigPTB** [comment]
If you consider _venvs_ as a bit too complicated. I would strongly advise you, whenever there issome spare time, to take a quick look at them and try out the mechanism. It is definitely best practice to use them and does not really compilcate things at all. Rather saves some pain in the long run with contradicting dependencies of different packages. And obviously it isolates environments, which helps pinpointing problems. Since I do not have a Windows machine at my hands, I cannot fully try to recreate your issue.

**Comment by BjoernLudwigPTB** [comment]
Maybe it is one of these strange Windows PYTHONPATH issues. I will try to get a hold of a Windows machine, to at least take one look at it myself. If you have more input, let me know!

The other/older [_pyxml2pdf_](https://github.com/kelvin0/PyXML2PDF) never was properly released, so this cannot intermingle with your situation. Besides, the `pip install` command outputs the correct version in the download part.

**Comment by Andde** [comment]
That was interesting, deleted the pyxml2pdf related info from site-packages, reinstalled with pip and now all works. Sorry for the inconvenience. After running pip again, shows 0.3.2 for the pyxml2pdf version also.

**Comment by BjoernLudwigPTB** [comment]
Uh, that's strange. Alright, thanks for letting me know! By the way, v0.3.3 just got released with official Python 3.9 and 3.10 support. Everything should work as expected.

If you have any questions or further remarks, please let me know!
